core/controllers/editor.py

```diff
@@ -854,14 +858,22 @@ def post(self, exploration_id):
                 'Expected a filename ending in .%s, received %s' %
                 (file_format, filename))
 
-        # Save the file.
-        fs = fs_domain.AbstractFileSystem(
-            fs_domain.ExplorationFileSystem(exploration_id))
-        if fs.isfile(filename):
+        mimetype = 'image/' + str(extension)
```

Comment:
Use 'image/%s' % extension. Also why do you need the cast to str?

Reply:
Yeah, we don't need to do that.